### PR TITLE
#955 Clean cmake targets to avoid invisible link between internal lib…

### PR DESCRIPTION
**Pull request by gautierbureau**
…raries.

### CODE REVIEW

**Comment by gautierbureau** [comment]
This is the part regarding cmake that I did in #478 previously. Changes in 478 will be bigger than the first proposition so I prefer to split it

**Comment by gautierbureau** [comment]
I forgot to build the tests hence the error

File: dynawo/3rdParty/nicslu/nicslu-chain.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+#
+# Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
```
Comment (rosiereflo) [inline dynawo/3rdParty/nicslu/nicslu-chain.sh:3]: 2020

File: dynawo/CMakeLists.txt [hunk 2]
```diff
@@ -88,6 +89,7 @@ set(CXX11_ENABLED ON CACHE BOOL "")
 if(MSVC)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -D_WIN32_WINNT=0x0601)
   add_definitions(-DLANG_CXX11)
+  add_definitions(-bigobj)
```
Comment (rosiereflo) [inline dynawo/CMakeLists.txt:92]: Did you encounter the maximum size issue?

File: dynawo/cmake/FindSundials.cmake [hunk 3]
```diff
@@ -65,6 +65,7 @@ if (SUNDIALS_INCLUDE_DIR AND SUNDIALS_IDA_LIBRARY)
     "int main() {\n"
     "  void* IDAMem = IDACreate();\n"
     "  int flag = IDASetMinStep(IDAMem, 0.1);\n"
+    "  static_cast<void>(flag);"
```
Comment (rosiereflo) [inline dynawo/cmake/FindSundials.cmake:68]: Why is this needed?

File: dynawo/sources/Common/CMakeLists.txt [hunk 4]
```diff
@@ -59,104 +59,95 @@ set(COMMON_SOURCES
...
-  )
-
-set(UTIL_SOURCES
-  DYNTimer.cpp
-  )
-
-set(UTIL_INCLUDE_HEADERS
+  DYNErrorQueue.h
+  DYNIoDico.h
   DYNTimer.h
```
Comment (rosiereflo) [inline dynawo/sources/Common/CMakeLists.txt:100]: Why are we removing the Dynawo_util library?

File: dynawo/sources/Modeler/Common/DYNDynamicData.cpp [hunk 5]
```diff
@@ -303,13 +301,9 @@ DynamicData::getParametersSet(const string& modelId, const string& parFile, cons
...
   parameters::XmlImporter parametersImporter;
 
-  try {
-    shared_ptr<ParametersSetCollection> parametersSetCollection = parametersImporter.importFromFile(canonical(parFile, rootDirectory_));
-    referenceParameters_[parFile] = parametersSetCollection;
-    return parametersSetCollection->getParametersSet(parId);
-  } catch (const xml::sax::parser::ParserException& exp) {
-    throw DYNError(Error::MODELER, XmlFileParsingError, parFile, exp.what());
-  }
+  shared_ptr<ParametersSetCollection> parametersSetCollection = parametersImporter.importFromFile(canonical(parFile, rootDirectory_));
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/DYNDynamicData.cpp:304]: Please keep the proper handle of this exception

File: dynawo/sources/Modeler/util/DYNGeneratePreassembled.cpp [hunk 6]
```diff
@@ -207,13 +205,10 @@ int main(int argc, char ** argv) {
       }
       throw DYNError(DYN::Error::MODELER, FileGenerationFailed, libList.c_str());
     }
-  } catch (const xml::sax::parser::ParserException& exp) {
-    Trace::error() << DYNLog(XmlParsingError, dydFileName, exp.what()) << Trace::endline;
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/util/DYNGeneratePreassembled.cpp:211]: Same as above.

File: dynawo/sources/ModelicaCompiler/compileCppModelicaModelInDynamicLib.cmake.in [hunk 7]
```diff
@@ -111,6 +111,15 @@ find_package(Boost 1.64 REQUIRED
 set(SUITESPARSE_HOME "$ENV{DYNAWO_SUITESPARSE_INSTALL_DIR}")
 find_package(SuiteSparse REQUIRED)
 
+if(DEFINED ENV{DYNAWO_NICSLU_INSTALL_DIR})
```
Comment (rosiereflo) [inline dynawo/sources/ModelicaCompiler/compileCppModelicaModelInDynamicLib.cmake.in:114]: Do we really need the NICSLU when compiling models?

File: dynawo/sources/Solvers/AlgebraicSolvers/CMakeLists.txt [hunk 8]
```diff
@@ -31,27 +31,28 @@ target_include_directories(dynawo_SolverKINSOL
...
-)
+  )
 
 if(${NICSLU_FOUND})
   target_link_libraries(dynawo_SolverKINSOL
-    PUBLIC
+    PRIVATE
       Sundials::Sundials_SUNLINSOLNICSLU
-      Sundials::Sundials_SUNMATRIXSPARSE
-      NICSLU::NICSLU
```
Comment (rosiereflo) [inline dynawo/sources/Solvers/AlgebraicSolvers/CMakeLists.txt:53]: I am surprised that we remove NICSLU here as it is used in the library

Comment (gautierbureau) [reply dynawo/sources/Solvers/AlgebraicSolvers/CMakeLists.txt:53]: The target Sundials::Sundials_SUNLINSOLNICSLU carries the NICSLU dependency

File: dynawo/sources/Modeler/Common/DYNDynamicData.cpp (hunk 5 above)
Comment (gautierbureau) [reply dynawo/sources/Modeler/Common/DYNDynamicData.cpp:304]: The exception is already handled in the PAR API and will never be caught here

File: dynawo/sources/Modeler/util/DYNGeneratePreassembled.cpp (hunk 6 above)
Comment (gautierbureau) [reply dynawo/sources/Modeler/util/DYNGeneratePreassembled.cpp:211]: same as above

File: dynawo/sources/Common/CMakeLists.txt (hunk 4 above)
Comment (gautierbureau) [reply dynawo/sources/Common/CMakeLists.txt:100]: With the current implementation it is useless, it will be reintroduced with the changes for the singleton but it easier to delete it for the time being as it should never as existed after windows port

File: dynawo/cmake/FindSundials.cmake (hunk 3 above)
Comment (gautierbureau) [reply dynawo/cmake/FindSundials.cmake:68]: to use the flag variable and avoid a warning as error in some cases

File: dynawo/CMakeLists.txt (hunk 2 above)
Comment (gautierbureau) [reply dynawo/CMakeLists.txt:92]: No but it prevents any issue in the future

Comment (rosiereflo) [reply dynawo/CMakeLists.txt:92]: The recommendation is to use it on a need-to basis as it may create issue with old linkers. So I propose to not add it unless we need it.

Comment (gautierbureau) [reply dynawo/CMakeLists.txt:92]: ok

File: dynawo/sources/ModelicaCompiler/compileCppModelicaModelInDynamicLib.cmake.in (hunk 7 above)
Comment (gautierbureau) [reply dynawo/sources/ModelicaCompiler/compileCppModelicaModelInDynamicLib.cmake.in:114]: ModelManager depends on dynawo_SolverKINSOL which depends on NICLSU
